### PR TITLE
feat(metadata): provide db version

### PR DESCRIPTION
**Pull request by grieve54706**
Add new API `/v2/connector/{data_source}/metadata/version` to provide the DB version.
The request body follows other metadata APIs.

### CODE REVIEW

File: ibis-server/tests/routers/v2/connector/test_clickhouse.py [hunk 1]
```diff
@@ -529,6 +529,15 @@ def test_metadata_list_constraints(clickhouse: ClickHouseContainer):
...
         assert len(result) == 0
 
+    def test_metadata_db_version(clickhouse: ClickHouseContainer):
+        connection_info = _to_connection_info(clickhouse)
+        response = client.post(
+            url=f"{base_url}/metadata/version",
+            json={"connectionInfo": connection_info},
+        )
+        assert response.status_code == 200
+        assert response.text is not None
```
Comment (goldmedal) [inline ibis-server/tests/routers/v2/connector/test_clickhouse.py:539]: Is it possible to assert the version text? or is it not a static value?

File: ibis-server/tests/routers/v2/connector/test_trino.py [hunk 2]
```diff
@@ -402,6 +402,15 @@ def test_metadata_list_constraints(trino: TrinoContainer):
...
         assert len(result) == 0
 
+    def test_metadata_db_version(trino: TrinoContainer):
+        connection_info = _to_connection_info(trino)
+        response = client.post(
+            url=f"{base_url}/metadata/version",
+            json={"connectionInfo": connection_info},
+        )
+        assert response.status_code == 200
+        assert response.text is not None
```
Comment (goldmedal) [inline ibis-server/tests/routers/v2/connector/test_trino.py:412]: Same as the issue for clickhouse.

File: ibis-server/tests/routers/v2/connector/test_clickhouse.py (hunk 1 above)
Comment (grieve54706) [reply ibis-server/tests/routers/v2/connector/test_clickhouse.py:539]: We use the `head` tag. It is not a static version.

File: ibis-server/tests/routers/v2/connector/test_trino.py (hunk 2 above)
Comment (grieve54706) [reply ibis-server/tests/routers/v2/connector/test_trino.py:412]: We use the `latest` tag. It is not a static version.

**Review by goldmedal — APPROVED** [review]
Thanks @grieve54706 -- It makes sense to me. 👍

**Pull request merged**